inc/Pages/ProfileSettings.php

```diff
@@ -44,7 +49,53 @@ public function register() {
 	}
 
 	/**
-	 * Add fields to reqistration area and account area
+	 * Generates correctly formatted woocommerce hook based of plugin settings.
+	 *
+	 * @return string
+	 */
+	public static function generate_checkbox_location() {
+		$settings = DataHandler::get_smaily_results()['result'];
+
+		$order    = $settings['checkbox_order'];
+		$location = $settings['checkbox_location'];
+
+		// Syntax - woocommerce_before_checkout_billing_form.
+		$location = 'woocommerce_' . $order . '_' . $location;
+
+		return $location;
+	}
+
+	/**
+	 * Add newsletter subscribe button to admin preferred place in checkout page.
+	 *
+	 * @return void
+	 */
+	public function smaily_checkout_newsletter_checkbox() {
+		$settings = DataHandler::get_smaily_results()['result'];
+		$display  = $settings['checkbox_display'];
+		?>
+		<p
+		class="form-row form-row-wide"
+		id="smaily-checkout-subscribe"
+		style=<?php echo( 'hidden_unchecked' === $display ? '"display:none;"' : '""' ); ?>>
+			<span class="woocommerce-input-wrapper">
+				<label for="user_newsletter" class="checkbox">
+					<input
+					type="checkbox"
+					class="input-checkbox"
+					name="user_newsletter"
+					id="user_newsletter"
+					value="1"
+					<?php echo( 'visible_checked' === $display ? 'checked' : '' ); ?>>
+					<?php esc_html_e( 'Subscribe to newsletter', 'smaily' ); ?>
+				</label>
+			</span>
+		</p>
```

Comment:
Plugin uses templates to render RSS feed and administrative form. Isn't it possible to use templating here also?

Reply:
I think it's not possible or would be unnecessarily complex because of the [hook system](https://github.com/woocommerce/woocommerce/blob/bdf77f5648fb44b86b19ac4f8a01536b1ee6baa1/templates/checkout/form-billing.php#L44) Wordpress has. I found a better solution by using `woocommerce_form_field()` function.

Reply:
https://github.com/sendsmaily/smaily-woocommerce-plugin/pull/45/commits/bcd2c5a17facf4ecfd78c4b50b7074c44337d820